### PR TITLE
Update: Checkpoint-template updated with last version feats (v.18)

### PR DESCRIPTION
**Pull request by antiyro**
Updated Starknet version + ORM feats.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,14 +9,15 @@
     "build": "tsc -p tsconfig.build.json",
     "dev": "nodemon src/index.ts",
     "start": "node dist/src/index.js",
-    "test": "jest"
+    "test": "jest",
+    "codegen": "node ./node_modules/.bin/checkpoint generate"
```
Comment (bonustrack) [inline package.json:13]:
```suggestion
    "generate": "node ./node_modules/.bin/checkpoint generate"
```

File: package.json [hunk 2]
```diff
@@ -9,14 +9,15 @@
     "build": "tsc -p tsconfig.build.json",
     "dev": "nodemon src/index.ts",
     "start": "node dist/src/index.js",
```
Comment (bonustrack) [inline package.json:11]: We should add "generate" script within start otherwise build or deployment would fail.

File: src/writers.ts [hunk 3]
```diff
@@ -18,8 +19,8 @@ export async function handleNewPost({ block, tx, event, mysql }: Parameters<Chec
   let tag = '';
   const contentLength = BigInt(event.data[1]);
   const tagLength = BigInt(event.data[2 + Number(contentLength)]);
-  const timestamp = block.timestamp;
-  const blockNumber = block.block_number;
+  const timestamp = block!.timestamp;
```
Comment (bonustrack) [inline src/writers.ts]: Why do you need the `!`, this return warning from GitHub Actions

**Comment by bonustrack** [comment]
Got this error when trying to run the build script, can you reproduce?
image

Seem to happen with GitHub action too, it might be an issue from Checkpoint directly:
https://github.com/checkpoint-labs/checkpoint-template/actions/runs/5319980183/jobs/9633146693?pr=5

File: package.json (hunk 1 above)
Comment (foxweba3) [reply package.json:13]: https://github.com/checkpoint-labs/checkpoint-template/pull/5#discussion_r1229431129

**Comment by bonustrack** [comment]
Moving to #7

**Pull request closed without merging**